ochk/structures_custom_service_test.go

```diff
@@ -0,0 +1,65 @@
+package ochk
+
+import (
+	"github.com/chmurakrajowa/terraform-provider-ochk/ochk/sdk/gen/models"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestFlattenExpandCustomServicePorts(t *testing.T) {
+	// FIXME problem on equal
+	t.Skip()
+	cases := []struct {
+		expanded  []*models.L4PortSetEntry
+		flattened []map[string]interface{}
+	}{
+
+		// nil values
+		{
+			expanded:  nil,
+			flattened: nil,
+		},
+
+		{
+			expanded: []*models.L4PortSetEntry{
+				{
+					L4PortSetEntryID: "id",
+					L4Protocol:       "protocol",
+					SourcePorts:      transformInterfaceSliceToStringSlice(flattenStringSlice([]string{"1", "2", "3"}).List()),
+					DestinationPorts: transformInterfaceSliceToStringSlice(flattenStringSlice([]string{"3", "4", "5"}).List()),
+				},
+				{
+					L4PortSetEntryID: "id2",
+					L4Protocol:       "protocol2",
+					SourcePorts:      transformInterfaceSliceToStringSlice(flattenStringSlice([]string{"11", "22", "33"}).List()),
+					DestinationPorts: transformInterfaceSliceToStringSlice(flattenStringSlice([]string{"33", "44", "55"}).List()),
+				},
+			},
+			flattened: []map[string]interface{}{
+				{
+					"id":          "id",
+					"protocol":    "protocol",
+					"source":      flattenStringSlice([]string{"1", "2", "3"}),
+					"destination": flattenStringSlice([]string{"3", "4", "5"}),
+				},
+				{
+					"id":          "id2",
+					"protocol":    "protocol2",
+					"source":      flattenStringSlice([]string{"11", "22", "33"}),
+					"destination": flattenStringSlice([]string{"33", "44", "55"}),
+				},
+			},
+		},
+	}
+
+	for _, c := range cases {
+		flattenedList := mapSliceToInterfaceSlice(c.flattened)
+		outFlattened := flattenCustomServicePorts(c.expanded)
+		//FIXME values are identical, but reported as not equal
```

Comment:
nie do końca jest to samo. Po jednej stronie masz []map[string]interface{} a po drugiej []interface{}

Reply:
Nie. Po obu stronach jest []interface{}. Diff w outpucie jest zgłoszony ale jak porównam to jest identyczny. Sprawdź sam pls, coś dziwnego.